x/delegation/types/keys.go

```
@@ -82,55 +109,95 @@ func ParseStakerAssetIDAndOperator(key []byte) (keys *SingleDelegationInfoReq, e
+	EpochNumber     uint64
+	UndelegationID  uint64
+}
+
+func GetPendingUndelegationRecordKey(epochIdentifier string, epochNumber int64, undelegationID uint64) []byte {
+	return utils.AppendMany(
+		// length of identifier,8bytes
+		sdk.Uint64ToBigEndian(uint64(len(epochIdentifier))),
+		// epoch identifier, length = len(epochIdentifier)
+		[]byte(epochIdentifier),
+		// Append the epoch number,8bytes
+		sdk.Uint64ToBigEndian(uint64(epochNumber)),
```

golangci-lint (1.62.2)

[high] G115: integer overflow conversion int64 -> uint64

(gosec)